sdk/src/types/block/context_input/mod.rs

```diff
@@ -32,40 +44,80 @@ impl ContextInput {
     /// Returns the context input kind of a `ContextInput`.
     pub fn kind(&self) -> u8 {
         match self {
+            Self::Commitment(_) => CommitmentContextInput::KIND,
+            Self::BlockIssuanceCredit(_) => BlockIssuanceCreditContextInput::KIND,
             Self::Reward(_) => RewardContextInput::KIND,
         }
     }
 
+    /// Checks whether the context input is a [`CommitmentContextInput`].
+    pub fn is_commitment(&self) -> bool {
+        matches!(self, Self::Commitment(_))
+    }
+
+    /// Checks whether the context input is a [`BlockIssuanceCreditContextInput`].
+    pub fn is_block_issuance_credit(&self) -> bool {
+        matches!(self, Self::BlockIssuanceCredit(_))
+    }
+
     /// Checks whether the context input is a [`RewardContextInput`].
     pub fn is_reward(&self) -> bool {
         matches!(self, Self::Reward(_))
     }
 
+    /// Gets the input as an actual [`CommitmentContextInput`].
+    /// PANIC: do not call on a non-commitment context input.
+    pub fn as_commitment(&self) -> &CommitmentContextInput {
+        if let Self::Commitment(input) = self {
+            input
+        } else {
+            panic!("context input is not of type commitment: {:?}", self);
```

Comment:
Please use the same panic message as for other enums https://github.com/iotaledger/iota-sdk/blob/2.0/sdk/src/types/block/output/mod.rs#L258